manifests/consul_exporter.pp

```diff
@@ -0,0 +1,163 @@
+#
+#  [*os*]
+#  Operating system (linux is the only one supported)
+#
+#  [*package_ensure*]
+#  If package, then use this for package ensure default 'latest'
+#
+#  [*package_name*]
+#  The binary package name - not available yet
+#
+#  [*purge_config_dir*]
+#  Purge config files no longer generated by Puppet
+#
+#  [*restart_on_change*]
+#  Should puppet restart the service on configuration change? (default true)
+#
+#  [*service_enable*]
+#  Whether to enable the service from puppet (default true)
+#
+#  [*service_ensure*]
+#  State ensured for the service (default 'running')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+#
+#  [*web_listen_address*]
+#  Address to listen on for web interface and telemetry. (default ":9107")
+#
+#  [*web_telemetry_path*]
+#  Path under which to expose metrics. (default "/metrics")
+class prometheus::consul_exporter (
+  $arch                  = $::prometheus::params::arch,
+  $bin_dir               = $::prometheus::params::bin_dir,
+  $consul_health_summary = $::prometheus::params::consul_exporter_consul_health_summary,
+  $consul_server         = $::prometheus::params::consul_exporter_consul_server,
+  $download_extension    = $::prometheus::params::consul_exporter_download_extension,
+  $download_url          = undef,
+  $download_url_base     = $::prometheus::params::consul_exporter_download_url_base,
+  $extra_groups          = $::prometheus::params::consul_exporter_extra_groups,
+  $extra_options         = '',
+  $group                 = $::prometheus::params::consul_exporter_group,
+  $init_style            = $::prometheus::params::init_style,
+  $install_method        = $::prometheus::params::install_method,
+  $log_level             = $::prometheus::params::consul_exporter_log_level,
+  $manage_group          = true,
+  $manage_service        = true,
+  $manage_user           = true,
+  $os                    = $::prometheus::params::os,
+  $package_ensure        = $::prometheus::params::consul_exporter_package_ensure,
+  $package_name          = $::prometheus::params::consul_exporter_package_name,
+  $purge_config_dir      = true,
+  $restart_on_change     = true,
+  $service_enable        = true,
+  $service_ensure        = 'running',
+  $user                  = $::prometheus::params::consul_exporter_user,
+  $version               = $::prometheus::params::consul_exporter_version,
+  $web_listen_address    = $::prometheus::params::consul_exporter_web_listen_address,
+  $web_telemetry_path    = $::prometheus::params::consul_exporter_web_telemetry_path,
+) inherits prometheus::params {
+  # Prometheus added a 'v' on the realease name at 0.3.0
+  if versioncmp ($version, '0.3.0') == -1 {
+    fail("I only support consul_exporter version '0.3.0' or higher")
+  }
+
+  $real_download_url = pick($download_url,"${download_url_base}/download/v${version}/${package_name}-${version}.${os}-${arch}.${download_extension}")
+  validate_bool($purge_config_dir)
```

Comment:
Hi @pjfbashton, could you replace those legacy functions with native puppet datatypes?